### ISSUE TITLE
How can I make nextjs supported optional chaining?

### ISSUE THREAD

**Issue by kuongknight** [zeit/next.js#9004]
I used nextjs 9.1 and I add custom .babelrc
`{
  "presets": ["next/babel"],
  "plugins": [
    ["@babel/plugin-proposal-optional-chaining"]
  ]
 }`
But I found complie issue: 
`Add @babel/plugin-proposal-nullish-coalescing-operator (https://git.io/vb4Se) to the 'plugins' section of your Babel config to enable transformation.`

**Comment by ijjk** [comment]
Hi, I'm not able to reproduce this. Adding `@babel/plugin-proposal-optional-chaining` to the `.babelrc`'s `plugins` like above on a minimal Next.js project appears to work fine with Next.js `v9.1.1`.

I'm going to close this, if you are still having difficulty, it's better to ask questions like this on our [Spectrum Thread](https://spectrum.chat/next-js) per our issue template

**Comment by aequasi** [comment]
This shouldn't be necessary anymore with TS 3.7 and Babel 7.7 anymore, right? Im still getting notices for this.

**Comment by iamsteeve** [comment]
Using the babel plugin is an option, but in theory it should work with Typescript 3.7.2 but it seems that this feature is taken by the babel compiler. Could someone with more experience explain us? :D o are we wrong with something?

**Comment by zenozen** [comment]
Correct me if I'm wrong. 

It seems that nextjs is still using babel internally to handle TS->JS. The `tsconfig.json` is only used for `fork-ts-checker-webpack-plugin` for type checking, and it doesn't emit any files. Hence a babelrc is needed probably for TS 3.7's new features.

Here's my .babelrc, seems to work fine:

```
{
  "presets": ["next/babel"],
  "plugins": [
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator"
  ]
}
```

**Comment by timneutkens** [comment]
Discussed with @Timer and we can add this by default for TypeScript. For JavaScript it's not in a stable stage yet so it's best to wait on it to be stable.

**Comment by rahul3103** [comment]
I have added in my project. But is this the right way?

`yarn add @babel/core babel-eslint @babel/plugin-proposal-nullish-coalescing-operator @babel/plugin-proposal-optional-chaining -D`

`.babelrc`
```
{
  "presets": ["next/babel"],
  "plugins": [
    ["styled-components", { "ssr": true, "displayName": true }],
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator"
  ]
}
```

`.eslintrc`
```
"parser": "babel-eslint",
```

**Comment by MAPESO** [comment]
> Using the babel plugin is an option, but in theory it should work with Typescript 3.7.2 but it seems that this feature is taken by the babel compiler. Could someone with more experience explain us? :D o are we wrong with something?

You are correct, the latest version of typescript supports **optional-chaining**.  In theory it is only necessary to use the most recent version to use this feature, the "problem" is that Next.js does not use `tsc` to transpile typescript. **Next is using Babel to transpile TypeScript** is the reason why it is not enough to have the latest version : )

If you want to use it you just need to create a `.babelrc` file with the following configuration

```json
{
    "presets": [
        "next/babel"
    ],
    "plugins": [
        "@babel/plugin-proposal-optional-chaining",
        "@babel/plugin-proposal-nullish-coalescing-operator"
    ]
}
```

**Comment by timneutkens** [comment]
Landed in latest stable Next.js.

**Comment by MAPESO** [comment]
@timneutkens Thank You for great job !

**Comment by JerryGreen** [comment]
Should this work out of the box? I have some `?.`, and it seems it's working perfectly locally, but upon deployment my deployment fails with an error (pointing out to exact place with `?.`):

```
05:50:40.242 | Failed to compile.
05:50:40.242 | ./src/pages/test-blog.tsx:19:11
05:50:40.242 | Type error: Expression expected.
```

**Comment by insivika** [comment]
Same here, I need to roll back my deployed node version in order to deploy on Vercel and this babel config doesnt seem to do the job:
```
{
    "presets": ["next/babel"],
    "plugins": ["@babel/plugin-proposal-optional-chaining"]
}

```

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
